src/directives/compose_area.ts

```diff
@@ -121,26 +126,63 @@ export default [
                                 $log.warn(logTag, 'Unhandled node:', node);
                         }
                     }
-                    const textMessageData: threema.TextMessageData = {text: text.trim()};
-                    // send as array
-                    return scope.submit('text', [textMessageData]);
+                    return new Promise((resolve, reject) => {
+                        let submitTexts = (strings: string[]) => {
+                            let messages: threema.TextMessageData[] = [];
+                            strings.forEach((piece: string) => {
+                                messages.push({
+                                    text: piece,
+                                } as threema.TextMessageData);
+                            });
+                            scope.submit('text', messages)
+                                .then(() => {
+                                    resolve();
+                                })
```

Comment:
I think this could be simplified to `.then(resolve)`. Same thing on the next line (`catch(reject)`) and line 160 (`}, reject);`).